Issue #73 feat(cli): --init command creates a yaml config file

ryzzaki opened the pull request:
Running the `--init` command will now create `.dxscannerrc.yaml` file IF it does not exist yet (this is a condition to prevent file overwriting). The file is created in the relevant project folder and pre-loaded with all available practices listed in DXS (all practices are commented out with `#`) with their relevant metadata - this logic is not hardcoded, therefore, practice addition process for DX developers remains the same as the function pulls the metadata from the root container that is initially created.

ryzzaki commented:
@prokopsimek please review the changes and let me know whether this is an acceptable solution within the architecture of the project and if it correctly fits your specifications described in issue #73.

prokopsimek commented on src/init.configurator.ts:
```diff
@@ -0,0 +1,35 @@
+
+const createYAML = (yamlPath: string, container: Container) => {
+  let yamlInitContent = `# practices:`;
+  const initializedPractices: ContainerPracticeMetadata[] = container.getAll(Types.Practice);
```
I guess  you can use `IPracticeWithMetadata` interface.

```suggestion
  const initializedPractices = container.getAll<IPracticeWithMetadata[]>(Types.Practice);
```

prokopsimek commented on src/init.configurator.ts:
```diff
@@ -0,0 +1,35 @@
+  });
+};
+
+export interface ContainerPracticeMetadata {
```
you can use `IPracticeWithMetadata`

prokopsimek commented on src/init.configurator.ts:
```diff
@@ -0,0 +1,35 @@
+  const initializedPractices: ContainerPracticeMetadata[] = container.getAll(Types.Practice);
+  for (const practice of initializedPractices) {
+    const dataObject = practice.getMetadata();
+    yamlInitContent += `\n#    ${dataObject.id}: ${dataObject.impact}`;
```
Please, sort all practices alphabetically.

prokopsimek commented:
@ryzzaki It's exactly what the #73 says! Thanks :) I've added few comments.

adelkahomolova commented on src/init.configurator.ts:
```diff
@@ -0,0 +1,30 @@
+export const initialize = (scanPath: string, container: Container) => {
+  const yamlPath = `${scanPath}/.dxscannerrc.yaml`;
+  cli.action.start(`Initializing configuration: ${yamlPath}`);
+  // check if .dxscannerrc.yaml already exists
```
Check also if there are .dxscannerrc.yml or .dxscannerrc.json. See ConfigProvider.ts

prokopsimek replied:
Good catch! 👍 And the `.dxscannerrc` without an extension (for future purposes).

adelkahomolova replied:
I was also thinking about the file without extension but we don't check it in ConfigProvider either. So maybe we should :)

adelkahomolova replied:
I was thinking about this also and you're right. I checked ConfigProvider and didn't notice one condition, so we check it also there. :)

ryzzaki commented:
I have added the pre-conditioning statements to check for the files stated by @adelkahomolova and @prokopsimek and I have also added one more integration test after @prokopsimek fixed my horrible test suite. All the test suites are passing 58/58 and the branch is ready for a merge with master. I have re-requested a final review from @adelkahomolova

prokopsimek commented:
:tada: This PR is included in version 1.22.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/dx-scanner)
- [GitHub release](https://github.com/DXHeroes/dx-scanner/releases/tag/v1.22.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
